src/Cassandra/Requests/RequestExecution.cs

```diff
@@ -332,28 +337,27 @@ private void HandleException(Exception ex)
                 _parent.SetNoMoreHosts(exception, this);
                 return;
             }
-            var h = _host;
-            if (h != null)
-            {
-                _triedHosts[h.Address] = ex;
-            }
+
+            _triedHosts[host.Address] = ex;
             if (ex is OperationTimedOutException)
             {
                 RequestExecution.Logger.Warning(ex.Message);
                 var connection = _connection;
-                if (h == null || connection == null)
+                if (connection == null)
                 {
-                    RequestExecution.Logger.Error("Host and Connection must not be null");
+                    RequestExecution.Logger.Error("Connection must not be null");
                 }
                 else
                 {
                     // Checks how many timed out operations are in the connection
-                    _session.CheckHealth(h, connection);
+                    _session.CheckHealth(host, connection);
                 }
             }
-            var decision = RequestExecution.GetRetryDecision(
-                ex, _parent.RetryPolicy, _parent.Statement, _session.Cluster.Configuration, _retryCount);
-            switch (decision.DecisionType)
+
+            var retryInformation = GetRetryDecisionWithReason(
```

Comment:
Given that we have the `Unsent` property (connection error before send), we could blindly retry on the next host on those cases.

No need to address it now, we could tackle it in a separate ticket.

Reply:
Hmm we should probably need some more tests on that property before changing behavior based on it. At the moment it's only used to choose which metric counter to increment which is not that critical if the wrong counter is incremented. I'll create a ticket for that.